src/poetry/publishing/uploader.py

```diff
@@ -261,12 +261,12 @@ def _upload_file(
                         headers={"Content-Type": monitor.content_type},
                         timeout=REQUESTS_TIMEOUT,
                     )
-                if resp is None or 200 <= resp.status_code < 300:
+                if resp is None or 200 <= resp.status_code < 299:
```

Comment:
you have introduced an out by one error here and elsewhere (what happens for 299? etc)

Reply:
Thanks, fixed!